datadriven/src/sgpp/datadriven/algorithm/DBMatOnlineDE.cpp

```diff
@@ -493,6 +569,82 @@ std::vector<DataVector> DBMatOnlineDE::computeWeightedBFromBatchTwoDatasets(
   return std::vector<DataVector>(3);  // by default all three vectors are empty
 }
 
+DataVector DBMatOnlineDE::computeWeightedDerivativeBFromBatch(
+    DataMatrix& m, Grid& grid, DensityEstimationConfiguration& densityEstimationConfig,
+    bool weighted) {
+  if (m.getNrows() > 0) {
+    DataMatrix& lhsMatrix = offlineObject.getDecomposedMatrix();
+
+    // in case OrthoAdapt or both SMW_, the current size is not lhs size, but B size
+    bool use_B_size = false;
+    size_t B_size = 0;
+    sgpp::datadriven::DBMatOnlineDEOrthoAdapt* this_OrthoAdapt_pointer;
+    if (densityEstimationConfig.decomposition_ ==
+        sgpp::datadriven::MatrixDecompositionType::OrthoAdapt) {
+      this_OrthoAdapt_pointer = static_cast<sgpp::datadriven::DBMatOnlineDEOrthoAdapt*>(&*this);
+      if (this_OrthoAdapt_pointer->getB().getNcols() > 1) {
+        use_B_size = true;
+        B_size = this_OrthoAdapt_pointer->getB().getNcols();
+      }
+    }
+
+    sgpp::datadriven::DBMatOnlineDE_SMW* this_SMW_pointer;
+    if (densityEstimationConfig.decomposition_ ==
+            sgpp::datadriven::MatrixDecompositionType::SMW_ortho ||
+        densityEstimationConfig.decomposition_ ==
+            sgpp::datadriven::MatrixDecompositionType::SMW_chol) {
+      this_SMW_pointer = static_cast<sgpp::datadriven::DBMatOnlineDE_SMW*>(&*this);
+      if (this_SMW_pointer->getB().getNcols() > 1) {
+        use_B_size = true;
+        B_size = this_SMW_pointer->getB().getNcols();
+      }
+    }
+
+    // Compute right hand side of the equation:
+    size_t numberOfPoints = m.getNrows();
+    DataVector b(use_B_size ? B_size : lhsMatrix.getNcols());
+
+    if (b.getSize() != grid.getSize()) {
+      throw sgpp::base::algorithm_exception(
+          "In DBMatOnlineDE::computeWeightedBFromBatch: b doesn't match size of system matrix");
+    }
+
+    /*
+    std::unique_ptr<sgpp::base::OperationMultipleEval> B(
+        (offlineObject.interactions.size() == 0)
+            ? sgpp::op_factory::createOperationMultipleEval(grid, m)
+            : sgpp::op_factory::createOperationMultipleEvalInter(grid, m,
+                                                                 offlineObject.interactions));
+    */
```

Comment:
There is a bunch of commented code in this file. This should be removed if it doesn't have a purpose anymore

Reply:
All the commented pieces of code (also the ones you've marked below) serve as basis for extending these new approaches with features that exist for the regular density estimation. Things like geometry aware grids (what is here, this MultipleEvalInter), enabling the visualization module, enabling hyperparameter optimization etc. 
I wanted this PR to be the basic implementation of the algorithm, and then in a next PR I will go through all these features and see which of them make sense to be implemented for each new method or not. And it's easier and safer to have them commented out first and then I can remove things, so that I don't forget something :) If you think these commented lines will increase the git history too much, I can start with adding the features now in this pull request instead of a separate one that weeds them out.